fraud/service.go

```diff
@@ -88,20 +117,19 @@ func (f *service) processIncoming(
 		}
 		return pubsub.ValidationReject
 	}
+	// we can skip an incoming proof if it has been already stored locally
+	if f.verifyLocal(ctx, proofType, hex.EncodeToString(proof.HeaderHash()), msg.Data) {
+		return pubsub.ValidationIgnore
+	}
```

Comment:
Nide! Should we do the same for the requesting part? Meaning that its possible that fifth Syncer request will get a FP which you've just recevied from the Pubsub

Reply:
No, because process incoming is called anyway